src/cpp/fastdds/domain/DomainParticipant.cpp

```diff
@@ -59,8 +59,10 @@ ReturnCode_t DomainParticipant::enable()
         return ReturnCode_t::RETCODE_OK;
     }
 
-    Entity::enable();
-    return impl_->enable();
+    enable_ = true;
```

Comment:
This should not be necessary, we are setting the value after enabling the implementation.

Reply:
Yes, it is. Inside `impl_->enable()`, we may call the `enable()` method on previously created entities, which will fail if the participant is not enabled.